### ISSUE TITLE
Full support for Vector API in GraalVM

### ISSUE THREAD

**Issue by alina-yur** [oracle/graal#10285]
### TL;DR

Implement [the Vector API (JEP 469)](https://openjdk.org/jeps/469) in GraalVM JIT and Native Image.

The [Vector API](https://openjdk.org/jeps/469) enables vector computations that reliably compile at runtime to optimal vector instructions, resulting in performance superior to equivalent scalar computations. Currently, the API is in the eighth incubator in OpenJDK 23 and will incubate until necessary features of Project Valhalla become available as preview features. 
Our goal is to add full Vector API support in GraalVM, both Graal JIT and Native Image.

### Goals

- Full support of Vector API in Graal JIT. In GraalVM, initial optimization of Vector API operations was introduced in [GraalVM for JDK 21](https://www.graalvm.org/release-notes/JDK_21/), enabled by default. It included support for load, store, basic arithmetic, reduce, compare, and blend operations. Our goal is full support, with a major update in the upcoming GraalVM for JDK 24 release (March 2024).
- Full support of Vector API in GraalVM Native Image. Currently, this is work in progress; can be tested in Oracle GraalVM for JDK 24+ with `-H:+VectorAPISupport`).
- Ensure high performance when running on GraalVM.
- Supported architectures: supported on both AArch64 and AMD64.

### Non-goals

- Perform autovectorization of loops when no Vector API is used

### Related PRs

- https://github.com/oracle/graal/pull/9781
- https://github.com/oracle/graal/pull/9677

**Comment by RainerXE** [comment]
I hope this covers MacOS as well 😉

**Comment by gergo-** [comment]
Support for Vector API optimization in GraalVM Community Edition landed in https://github.com/oracle/graal/pull/11255 and will be part of the GraalVM for JDK 25 release in September. This covers Vector API optimization both on JIT and on Native Image.
